app/src/lib/stores/app-store.ts

```diff
@@ -3088,14 +3088,7 @@ export class AppStore extends TypedBaseStore<IAppState> {
       }
     }
 
-    const { tip } = gitStore
-    await gitStore.performFailableOperation(() => gitStore.merge(branch), {
```

Comment:
is this a pattern we'd like to avoid in the future @shiftkey ? Its pretty prevalent in app-store currently, so i think its worth making a `tech-debt` issue to track and document it

Reply:
The pattern I'm trying to avoid here is "wrapping a GitStore function in error handling" because `GitStore` already has this responsibility, and I think we should internalize it whenever appropriate.

Looking over the other usage in `AppStore`, there seem to be some functions that don't cache anything (so they bypass `GitStore` and instead invoke the Git action directly) but need error handling, which is likely why we made `GitStore.performFailableOperation` public. 

I'd love to get to a point where our error handling lives outside `GitStore` so we can compose things together better, but for now this is a side-effect of our Flux architecture (because errors currently flow through `GitStore -> AppStore -> Dispatcher` before reaching our error handlers).

Reply:
> I'd love to get to a point where our error handling lives outside GitStore so we can compose things together better

@outofambit I noodled on a spike of what this might look like as a way to tidy things up - let me know if this helps with your understanding of it and we can jam on this later: https://github.com/desktop/desktop/compare/experiment-isolate-error-handling